internal/pkg/mock/backend.go

```diff
@@ -1,29 +1,45 @@
 package mock
 
 import (
+	"errors"
+
 	"github.com/puppetlabs/prm/pkg/prm"
 )
 
 type MockBackend struct {
 	StatusIsAvailable   bool
 	StatusMessageString string
+	ToolAvalible        bool
+	ExecReturn          string
 }
 
 func (m *MockBackend) Status() prm.BackendStatus {
 	return prm.BackendStatus{IsAvailable: m.StatusIsAvailable, StatusMsg: m.StatusMessageString}
 }
 
-// Implement when needed
 func (m *MockBackend) GetTool(tool *prm.Tool, prmConfig prm.Config) error {
-	return nil
+	if m.ToolAvalible {
+		return nil
+	} else {
+		return errors.New("Tool Not Found")
+	}
 }
 
 // Implement when needed
 func (m *MockBackend) Validate(tool *prm.Tool) (prm.ToolExitCode, error) {
 	return prm.FAILURE, nil
 }
 
-// Implement when needed
 func (m *MockBackend) Exec(tool *prm.Tool, args []string, prmConfig prm.Config, paths prm.DirectoryPaths) (prm.ToolExitCode, error) {
-	return prm.FAILURE, nil
+	if m.ExecReturn == "SUCCESS" {
+		return prm.SUCCESS, nil
+	} else if m.ExecReturn == "FAILURE" {
+		return prm.FAILURE, nil
+	} else if m.ExecReturn == "TOOL_ERROR" {
+		return prm.TOOL_ERROR, nil
+	} else if m.ExecReturn == "TOOL_NOT_FOUND" {
+		return prm.TOOL_NOT_FOUND, nil
+	} else {
+		return prm.FAILURE, errors.New("DOCKER FAILURE")
+	}
```

Comment:
I think we can more idiomatically use a switch statement here, like this:

https://github.com/puppetlabs/prm/blob/main/pkg/prm/exec.go#L33-L47